enhancements/update/available-update-metadata.md

```diff
@@ -0,0 +1,215 @@
+status: implementable
+---
+
+# Available-update Metadata
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Currently the ClusterVersion API has [a single `Update` type][api-update] used for both [the `desiredUpdate` input][api-desiredUpdate] and the [`desired`][api-desired] and [`availableUpdates`][api-availableUpdates] outputs.
+But update requests like `desiredUpdate` have different schema requirements than output like `desired` and `availableUpdates`.
+For example, [the `force` property][api-force] makes sense in the `desiredUpdate` context, but not in the `availableUpdates` contexts.
+And release metadata like landing page URIs and available channels makes sense in the `desired` and `availableUpdates` contexts, but not in the `desiredUpdate` context.
+This enhancement adds a new `Release` type to be used in `status` properties so we can distinguish between input and output schema.
+
+## Motivation
+
+Cincinnati provides [metadata about releases][cincinnati-metadata], including (since [here][cincinnati-channel-metadata]) the channels to which the release currently belongs.
+Exposing that information to in-cluster consumers allows them to use live metadata, instead of talking to Cincinnati directly or hard-coding expected values.
+For example, the console currently [hard-codes an expected set of available channels][console-channels] for a given release; with this enhancement they could look up available channels in `status.desired.metadata["io.openshift.upgrades.graph.release.channels"]`.
+
+### Goals
+
+* Exposing release metadata provided by upstream Cincinnati for the currently-desired and available-update releases.
+* Documenting some well-known metadata keys.
+
+### Non-Goals
+
+* Restricting metadata keys to those documented in this enhancement.
+    Cincinnati may continue to add and remove keys as they see fit as long as they provide at least the keys required by this enhancement.
+* Teaching Cincinnati to support requests without `channel` query parameters.
+    This might be useful for recovering from users specifying [`desiredUpdate`][api-desiredUpdate] that were not in their configured [`channel`][api-channel], but I am fine leaving recovery to the users.
+    They should have received sufficient information to adjust their channel from wherever they received the target release they are supplying.
+
+## Proposal
+
+FIXME: In response to [Abhinav's comment](https://github.com/openshift/enhancements/pull/123#issuecomment-603370388), here are some alternative ways to do this.
+Once we reach a consensus on [*a*](#a) or [*b*](#b), the alternative will move into [the *alternatives* section](#alternatives)
```

Comment:
i think the current approach of updating the current CV status to use Release for availableUpdates list seems like a acceptable near term solution. we can look into a separate object later on.

The only thing we need to make sure is that Serialized API needs to stay the same for backward compatibilty we need to make sure Release is backward-compatible replacement for Update.
we need to make explicit call-out/decision to drop fields like force because they have never been set or used by anybody.

Reply:
> i think the current approach of updating the current CV status to use Release for availableUpdates list seems like a acceptable near term solution

I've punted "separate objects" to the *Alternatives* section with ca79c67 -> 9e9e0f5.

> we need to make explicit call-out/decision to drop fields like force because they have never been set or used by anybody.

Isn't that covered [here][1]?  Is there any wording you'd like to see added/altered there?

[1]: https://github.com/openshift/enhancements/blob/9e9e0f511939f1e3b7c45e9dea2c60d569863481/enhancements/update/available-update-metadata.md#removing-a-deprecated-feature